src/celeritas/ext/GeantSurfacePhysicsLoader.cc

```diff
@@ -0,0 +1,435 @@
+
+        {G4OpticalSurfaceFinish::Rough_LUT, "Rough_LUT"},
+        {G4OpticalSurfaceFinish::RoughTeflon_LUT, "RoughTeflon_LUT"},
+        {G4OpticalSurfaceFinish::RoughESR_LUT, "RoughESR_LUT"},
+        {G4OpticalSurfaceFinish::RoughESRGrease_LUT, "RoughESRGrease_LUT"},
+
+        {G4OpticalSurfaceFinish::Polished_LUT, "Polished_LUT"},
+        {G4OpticalSurfaceFinish::PolishedTeflon_LUT, "PolishedTeflon_LUT"},
+        {G4OpticalSurfaceFinish::PolishedESR_LUT, "PolishedESR_LUT"},
+        {G4OpticalSurfaceFinish::PolishedESRGrease_LUT,
+         "PolishedESRGrease_LUT"},
+
+        {G4OpticalSurfaceFinish::Detector_LUT, "Detector_LUT"},
+    };
+    return to_cstring_impl.find(value)->second.c_str();
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with Geant4 geometry data.
+ */
+GeantSurfacePhysicsLoader::GeantSurfacePhysicsLoader()
+    : geo_(celeritas::geant_geo().lock())
+{
+    CELER_VALIDATE(geo_, << "global Geant4 geometry is not loaded");
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Populate surface physics data.
+ */
+inp::SurfacePhysics GeantSurfacePhysicsLoader::operator()()
+{
+    inp::SurfacePhysics result;
+
+    for (auto sid : range(SurfaceId(geo_->num_surfaces())))
+    {
+        auto const* g4log_surf = geo_->id_to_geant(sid);
+        CELER_ASSERT(g4log_surf);
+        auto* g4surf_prop = g4log_surf->GetSurfaceProperty();
+        CELER_ASSERT(g4surf_prop);
+        auto* g4opt_surf = dynamic_cast<G4OpticalSurface*>(g4surf_prop);
+        auto const* g4mpt = g4opt_surf->GetMaterialPropertiesTable();
+        CELER_ASSERT(g4mpt);
+        detail::GeantMaterialPropertyGetter get_property{*g4mpt};
+
+        try
+        {
+            result.names.insert({sid, g4opt_surf->GetName()});
+            this->insert_reflectivity(sid, *g4opt_surf, get_property, result);
+            this->insert_roughness(sid, *g4opt_surf, result);
+            this->insert_interaction(sid, get_property, *g4opt_surf, result);
+            this->insert_efficiency(sid, get_property, result);
+
+            // Ensure that data is not incompatible with selected model
+            this->validate_model(sid, *g4opt_surf, result);
+        }
+        catch (RuntimeError const& e)
+        {
+            CELER_VALIDATE(
+                false,
+                << "failed to convert surface " << g4opt_surf->GetName()
+                << " with model " << to_cstring(g4opt_surf->GetModel())
+                << " and finish " << to_cstring(g4opt_surf->GetFinish())
+                << ": " << e.details().which << ", " << e.details().what);
+        }
+    }
+
+    return result;
+}
+
+//---------------------------------------------------------------------------//
+// PRIVATE MEMBER FUNCTIONS
+//---------------------------------------------------------------------------//
+
+//---------------------------------------------------------------------------//
```

Comment:
```suggestion
//---------------------------------------------------------------------------//
```

There are a couple of these in the PR